### PR TITLE
Don't add elements to cache if they're found right away

### LINKED ISSUE #12: Found elements are not uncached
```
add element
wait for element
resolve promise

add element (different, but same selector)
wait for element
resolve promise (resolves with first element)
```

I'm adding a test

### PR DESCRIPTION
**Pull request by bfred-it**
Fixes #12

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -104,3 +104,15 @@ test('ensure different promises are returned on second call with the same select
...
+
+test.failing('ensure different promises are returned on second call with the same selector when first was found', async t => {
+	const prependElement = () => {
+		const el = document.createElement('p');
+		el.className = 'unicorn';
+		document.body.prepend(el);
+		return el;
+	};
+
+	t.is(prependElement(), await m('.unicorn'));
```
Comment (sindresorhus) [inline test.js]: You need to remove the element before doing the second check.

Comment (bfred-it) [reply test.js]: I’m prepending it, it’s always the first in the document. And it seems to work

**Comment by bfred-it** [comment]
:tadaaaaaaaaaaaaaaaaaa:

It fails because on the second call Line 36 runs before the targetCache Map for `document` has been created at all, so it fails, but the error is caught... somewhere? I think?

**Comment by bfred-it** [comment]
Nah, actually it's a simple race condition. If the element is found in the first loop, `cleanCache` is called before synchronously, **before** line 46 actually adds it.

PR WIP

**Comment by bfred-it** [comment]
Up, I always run into this (caught) errors while debugging RGH

**Comment by sindresorhus** [comment]
@bfred-it Is this PR done? I'm asking because you marked it `PR WIP`.

**Comment by bfred-it** [comment]
Yes, then I sent a couple commits and forgot to leave a comment 🙂

**Pull request merged**